nx/lib/nx/defn/kernel.ex

```diff
@@ -1488,111 +1421,143 @@ defmodule Nx.Defn.Kernel do
   """
   def assert_keys(keyword, keys) when Kernel.and(is_list(keyword), is_list(keys)) do
     for key <- keys, Kernel.not(Keyword.has_key?(keyword, key)) do
-      raise ArgumentError,
-            "expected key #{inspect(key)} in keyword list, got: #{inspect(keyword)}"
+      Kernel.raise(
+        ArgumentError,
+        "expected key #{Kernel.inspect(key)} in keyword list, got: #{Kernel.inspect(keyword)}"
+      )
     end
 
     keyword
   end
 
   @doc """
-  Asserts the `tensor` has a certain `shape`.
+  Raises a runtime exception with the given `message`.
 
-  If it succeeds, it returns the given tensor. Raises
-  an error otherwise.
+  See `raise/2` for more information on exceptions inside `defn`.
+  """
+  defmacro raise(message) do
```

Comment:
```suggestion
  # It needs to be a macro so we don't add stacktrace entries.
  # Since there is no defdelegate for macros, we do it manually.
  defmacro raise(message) do
```